far/common/2d/segment.hpp

```diff
@@ -0,0 +1,93 @@
+﻿#ifndef SEGMENT_HPP_DC38F7F7_0E1B_486E_8368_2ACE6415D234
+#define SEGMENT_HPP_DC38F7F7_0E1B_486E_8368_2ACE6415D234
+#pragma once
+
+/*
+segment.hpp
+
+*/
+/*
+Copyright © 2018 Far Group
```

Comment:
2018 -> 2025.

Why is it in 2d? Isn't it 1-dimentional conceptually?

Reply:
> 2018 -> 2025.

Ough! Sure, will fix it.

Yes, it is technically 1d. I could not think of a good place for it. Does it make sense to move it up to "common"?

Reply:
Yes, why not.